webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -299,6 +376,152 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
     };
   }
 
+  private parseXmlForm(form): Document | undefined {
```

Comment:
I'd move as much of this sort of stuff as is possible to a lib/file so it can be unit tested easily and then just called internally. Not sure if `private` is needed here. 

(I know this was just a conceptual prototype not productionised, but for forward looking feedback.)